client-src/default/index.js

```diff
@@ -226,14 +227,20 @@ const socketUrl = url.format({
   protocol,
   auth: urlParts.auth,
   hostname,
-  port,
+  // If sockPort is provided it'll be passed in via the __resourceQuery as a
+  // query param so it has to be parsed out of the querystring in order for the
+  // client to open the socket to the correct port.
+  port:
+    urlParts.path == null || urlParts.path === '/'
+      ? port
+      : querystring.parse(urlParts.path).sockPort || port,
   // If sockPath is provided it'll be passed in via the __resourceQuery as a
   // query param so it has to be parsed out of the querystring in order for the
   // client to open the socket to the correct location.
   pathname:
     urlParts.path == null || urlParts.path === '/'
       ? '/sockjs-node'
-      : querystring.parse(urlParts.path).sockPath || urlParts.path,
+      : querystring.parse(urlParts.path).sockPath || '/sockjs-node',
```

Comment:
I think it is breaking change, why you change this? I think it should be `urlParts.path` and for port `urlParts.port`

Reply:
because if you add sockPort and not add sockPath
the urlParts.path is '/&sockPort=8080' !== '/'
so the expression's result is urlParts.path is error

Reply:
Make sense :+1:

Reply:
> I think it is breaking change, why you change this? I think it should be `urlParts.path` and for port `urlParts.port`

ok, i know your means is sockPort use urlParts.port,it's my problem, i will modify it now,
and i do someting problem in accept cla

Reply:
CLA is ok:
```
licence/cla — Contributor License Agreement is signed.
```

Reply:
already modify it
now sockPort use urlParts.port
i am sorry for don't understand the design wealthy

Reply:
> I think it is breaking change, why you change this? I think it should be `urlParts.path` and for port `urlParts.port`

i having modify it
and now urlparts.port is options.sockPort || options.port
i am sorry for don't understand the design better

Reply:
> CLA is ok:
> 
> ```
> licence/cla — Contributor License Agreement is signed.
> ```

oh,i find the __resourceQuery is ?http://localhost8080&sockPath=xxx
and url.parse(__resourceQuery.substr[1]) cannont get the port directly
so only we can use ?http://localhost8080&sockPath=xxx&sockPort=xxx
and then by querystring.parse(urlparts.path) can get the sockPort
so the method which i use maybe true

Reply:
> http://localhost8080&sockPath=xxx&sockPort=xxx

Right solution :+1:

Reply:
I think we need return `urlParts.path` here :thinking:

Reply:
> I think we need return `urlParts.path` here 🤔

what is your idea?

Reply:
We should return `querystring.parse(urlParts.path).sockPath || urlParts.path`

Reply:
> We should return `querystring.parse(urlParts.path).sockPath || urlParts.path`

but if we dont add sockPath
now the urlParts.path is '/'
and the pathname is '/'

Reply:
To not change the previous behavior. We refactor this in future.

Reply:
ok，the cla is ok?
and what else should i do

Reply:
CLA is ok, just fix it and we merge this in near future, thanks for PR

Reply:
thanks for your reply because it's so late

Reply:
> CLA is ok, just fix it and we merge this in near future, thanks for PR

already use urlParts.path

Reply:
> CLA is ok, just fix it and we merge this in near future, thanks for PR

why createConfig.js not have 
if (argv.sockPath) {
    options.sockPath = argv.sockPath;
  }
it is a bug?

Reply:
No, is for CLI

Reply:
yeah, [at the pr](https://github.com/webpack/webpack-dev-server/pull/1797)